pyanaconda/modules/payload/dnf/packages/packages_interface.py

```diff
@@ -0,0 +1,220 @@
+#
+# DBus interface for packaging section.
+#
+# Copyright (C) 2019 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.dbus.property import emits_properties_changed
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+
+from pyanaconda.modules.common.constants.objects import DNF_PACKAGES
+from pyanaconda.modules.common.base import KickstartModuleInterfaceTemplate
+from pyanaconda.modules.payload.dnf.packages.constants import MultilibPolicy
+
+
+@dbus_interface(DNF_PACKAGES.interface_name)
+class PackagesHandlerInterface(KickstartModuleInterfaceTemplate):
+    """DBus interface for DNF packages sub-module."""
+
+    def connect_signals(self):
+        super().connect_signals()
+
+        self.implementation.core_group_enabled_changed.connect(self.changed("CoreGroupEnabled"))
+        self.implementation.environment_changed.connect(self.changed("Environment"))
+        self.implementation.groups_changed.connect(self.changed("Groups"))
+        self.implementation.packages_changed.connect(self.changed("Packages"))
+        self.implementation.excluded_groups_changed.connect(self.changed("ExcludedGroups"))
+        self.implementation.excluded_packages_changed.connect(self.changed("ExcludedPackages"))
+
+        self.implementation.docs_excluded_changed.connect(self.changed("DocsExcluded"))
+        self.implementation.weakdeps_excluded_changed.connect(self.changed("WeakdepsExcluded"))
+        self.implementation.missing_ignored_changed.connect(self.changed("MissingIgnored"))
+        self.implementation.languages_changed.connect(self.changed("Languages"))
+        self.implementation.multilib_policy_changed.connect(self.changed("MultilibPolicy"))
+        self.implementation.timeout_changed.connect(self.changed("Timeout"))
+        self.implementation.retries_changed.connect(self.changed("Retries"))
```

Comment:
Btw. we prefer to call the method `watch_property` now. The method `changed` will be removed.

Reply:
Will be solved later.